haystack/document_stores/base.py

```diff
@@ -381,7 +377,7 @@ def query_by_embedding_batch(
                     " as query_embs or a single filter that will be applied to each query_emb."
                 )
         else:
-            filters = [filters] * len(query_embs) if filters is not None else [{}] * len(query_embs)
```

Comment:
@tstadel I think we still need the `if filters is not None else [{}] * len(query_embs)` or something similar to it.
If filters is None, we otherwise run into a `NoneType is not iterable error`:

```python
>>> filters = None
>>> query_embs = [1,2,3]
>>> a = [filters] * len(query_embs)
>>> a
[None, None, None]
>>> for query_emb, filter in zip(query_embs, filters):
...     print(query_emb)
...     print(filter)
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
>>>
```

Reply:
This should be good as we do more like this:
```python
>>> filters = None
>>> query_embs = [1,2,3]
>>> filters = [filters] * len(query_embs)
>>> filters
[None, None, None]
>>> for query_emb, filter in zip(query_embs, filters):
...     print(query_emb)
...     print(filter)
...
1
None
2
None
3
None

```